### PR TITLE
chore(goversion): upgrade minimum version to 1.22

### PR DESCRIPTION
**Pull request by M4tteoP**
`Go 1.23` is out, so as per our policy, we should be fine supporting `go 1.22` and `go 1.23`:
Note: It should also unblock https://github.com/corazawaf/coraza/pull/1089, which requires go >= 1.22

### CODE REVIEW

File: internal/corazawaf/rule.go [hunk 1]
```diff
@@ -399,7 +399,7 @@ func (r *Rule) transformArg(arg types.MatchData, argIdx int, cache map[transform
 		default:
 			// NOTE: See comment on transformationKey struct to understand this hacky code
 			argKey := arg.Key()
-			argKeyPtr := (*reflect.StringHeader)(unsafe.Pointer(&argKey)).Data
+			argKeyPtr := (*reflect.StringHeader)(unsafe.Pointer(&argKey)).Data // nolint:staticcheck
```
Comment (M4tteoP) [inline internal/corazawaf/rule.go:402]: `reflect.StringHeader` has been deprecated and the linter is complaining.
Something like the following might do the trick, but I don't want to rush changes on something very hack and performance-related.
```
argKeyPtr2 := uintptr(unsafe.Pointer(unsafe.StringData(argKey)))
```

Comment (fzipi) [reply internal/corazawaf/rule.go:402]: Ugh, why do we need unsafe in the first place...

**Review by fzipi — APPROVED** [review]
I see this as a good change.

File: internal/corazawaf/rule.go (hunk 1 above)
Comment (M4tteoP) [reply internal/corazawaf/rule.go:402]: This is the historical context: https://github.com/corazawaf/coraza/pull/553. We should revisit and see if we have some other viable option. If this looks okay, we can go ahead with merging and I'll open an issue about that

Comment (fzipi) [reply internal/corazawaf/rule.go:402]: Yes, let's upgrade the golang part and create a follow up issue.

**Review by fzipi — COMMENTED** [review]
I'll try to get this done using renovate next time.

File: internal/corazawaf/rule.go (hunk 1 above)
Comment (M4tteoP) [reply internal/corazawaf/rule.go:402]: Filled https://github.com/corazawaf/coraza/issues/1147.

**Pull request merged**